src/URI.js

```diff
@@ -1863,6 +1863,7 @@
 
     if (resolved.path().charAt(0) !== '/') {
       basedir = base.directory();
+      basedir = basedir ? basedir : base.path().indexOf('/') === 0 ? '/' : '';
```

Comment:
yeah… this is personal preference, I agree.
since most of the code is using ternaries rather than "truthy-switches" (whatever they're called), I'd roll with your solution for now.